test/test_navierstokes.py

```diff
@@ -0,0 +1,352 @@
+        or eoc_rec0.max_error() < 1e-9
+    )
+    assert (
+        eoc_rec1.order_estimate() >= order - 0.5
+        or eoc_rec1.max_error() < 1e-9
+    )
+
+
+# Box grid generator widget lifted from @majosm and slightly bent
+def _get_box_mesh(dim, a, b, n, t=None):
+    dim_names = ["x", "y", "z"]
+    bttf = {}
+    for i in range(dim):
+        bttf["-"+str(i+1)] = ["-"+dim_names[i]]
+        bttf["+"+str(i+1)] = ["+"+dim_names[i]]
+    from meshmode.mesh.generation import generate_regular_rect_mesh as gen
+    return gen(a=a, b=b, n=n, boundary_tag_to_face=bttf, mesh_type=t)
+
+
+@pytest.mark.parametrize("order", [2, 3])
+def test_poiseuille_rhs(actx_factory, order):
+    """Test the Navier-Stokes operator using a Poiseuille state.
+
+    This state should yield rhs = 0 to FP.  The test is performed for 1, 2,
+    and 3 dimensions, with orders 1, 2, and 3, with and without passive species.
+    """
+    actx = actx_factory()
+    dim = 2
+    tolerance = 1e-9
+
+    from pytools.convergence import EOCRecorder
+    eoc_rec = EOCRecorder()
+
+    base_pressure = 100000.0
+    pressure_ratio = 1.001
+    mu = 1.0
+    left_boundary_location = 0
+    right_boundary_location = 0.1
+    ybottom = 0.
+    ytop = .02
+    nspecies = 0
+    mu = 1.0
+    kappa = 0.0
+    spec_diffusivity = 0 * np.ones(nspecies)
+    from mirgecom.gas_model import GasModel, make_fluid_state
+    gas_model = GasModel(
+        eos=IdealSingleGas(),
+        transport=SimpleTransport(viscosity=mu, thermal_conductivity=kappa,
+                                  species_diffusivity=spec_diffusivity))
+
+    def poiseuille_2d(x_vec, eos, cv=None, **kwargs):
+        y = x_vec[1]
+        x = x_vec[0]
+        x0 = left_boundary_location
+        xmax = right_boundary_location
+        xlen = xmax - x0
+        p_low = base_pressure
+        p_hi = pressure_ratio*base_pressure
+        dp = p_hi - p_low
+        dpdx = dp/xlen
+        h = ytop - ybottom
+        u_x = dpdx*y*(h - y)/(2*mu)
+        p_x = p_hi - dpdx*x
+        rho = 1.0
+        mass = 0*x + rho
+        u_y = 0*x
+        velocity = make_obj_array([u_x, u_y])
+        ke = .5*np.dot(velocity, velocity)*mass
+        gamma = eos.gamma()
+        if cv is not None:
+            mass = cv.mass
+            vel = cv.velocity
+            ke = .5*np.dot(vel, vel)*mass
+
+        rho_e = p_x/(gamma-1) + ke
+        return make_conserved(2, mass=mass, energy=rho_e,
+                              momentum=mass*velocity)
+
+    initializer = poiseuille_2d
+
+    # for nel_1d in [4, 8, 12]:
+    for nfac in [1, 2, 4, 8]:
+
+        npts_axis = nfac*(12, 20)
+        box_ll = (left_boundary_location, ybottom)
+        box_ur = (right_boundary_location, ytop)
+        mesh = _get_box_mesh(2, a=box_ll, b=box_ur, n=npts_axis)
+
+        logger.info(
+            f"Number of {dim}d elements: {mesh.nelements}"
+        )
+
+        discr = EagerDGDiscretization(actx, mesh, order=order)
```

Comment:
Use this as a reference to update this according to `overintegration`: https://github.com/illinois-ceesd/mirgecom/blob/thg/flux-diff-temp-seeding/examples/pulse-mpi.py#L164-L177